pkg/tsdb/jaeger/client.go

```diff
@@ -92,6 +95,82 @@ func (j *JaegerClient) Operations(s string) ([]string, error) {
 	return operations, err
 }
 
+func (j *JaegerClient) Search(query *JaegerQuery) ([]TraceResponse, error) {
+	jaegerURL, err := url.Parse(j.url)
+	if err != nil {
+		return []TraceResponse{}, fmt.Errorf("failed to parse Jaeger URL: %w", err)
+	}
+	jaegerURL.Path = "/api/traces"
+
+	if query.Tags != "" {
+		tagMap := make(map[string]string)
+		pairs := strings.Split(query.Tags, " ")
+		for _, pair := range pairs {
+			kv := strings.SplitN(pair, "=", 2)
+			if len(kv) != 2 {
+				continue
+			}
+			tagMap[kv[0]] = kv[1]
+		}
+
+		marshaledTags, err := json.Marshal(tagMap)
+		if err != nil {
+			return []TraceResponse{}, fmt.Errorf("failed to convert tags to JSON: %w", err)
+		}
+
+		query.Tags = string(marshaledTags)
```

Comment:
Is mutation of original query here on purpose? Do we need it? If not, I would prefer not to mutate it.